src/app/feature/theme/services/theme.service.ts

```diff
@@ -1,18 +1,44 @@
 import { Injectable } from "@angular/core";
-import { IpcService } from "src/app/shared/services/ipc/ipc.service";
+import { BehaviorSubject } from "rxjs";
 import { LocalStorageService } from "src/app/shared/services/local-storage/local-storage.service";
-import { BASE_THEME, BUILT_IN_EDITABLE_THEMES } from "../built-in-themes";
-import { AppTheme } from "../theme.model";
+import { THEMES } from "src/theme/themes";
 
 @Injectable({
   providedIn: "root",
 })
 export class ThemeService {
-  static THEME_UPDATE_CHANNEL = "THEME_UPDATE_CHANNEL";
+  currentTheme$ = new BehaviorSubject<string>(null);
+  availableThemes = THEMES;
 
-  currentTheme: AppTheme;
+  constructor(private localStorageService: LocalStorageService) {}
 
-  constructor(private ipcService: IpcService, private localStorageService: LocalStorageService) {}
+  init() {
+    const currentThemeName = this.getCurrentTheme();
+    if (currentThemeName) {
+      this.setTheme(currentThemeName);
+    } else {
+      this.setTheme("default");
+    }
+  }
+
+  public setTheme(themeName: string) {
+    if (this.availableThemes.includes(themeName)) {
+      document.body.dataset.theme = themeName;
+      this.currentTheme$.next(themeName);
+      // Use local storage so that the current theme persists across app launches
+      this.localStorageService.setString("currentTheme", themeName);
+    } else {
+      console.error(`No theme found with name "${themeName}"`);
+    }
+  }
+
+  public getCurrentTheme() {
+    return this.localStorageService.getString("currentTheme");
```

Comment:
nit(blocking) - we should prefix this with an `_` to mark as a reserved variable